iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java

```diff
@@ -91,8 +93,12 @@
 @SuppressWarnings("java:S1172")
 public class TreeFactory {
 
-  public FileTree file(Optional<List<InstructionTree>> instructions, Optional<SyntaxToken> spacing, SyntaxToken eof) {
-    return new FileTreeImpl(instructions.or(Collections.emptyList()), eof);
+  public FileTree file(Optional<List<DockerImageTree>> dockerImages, Optional<SyntaxToken> spacing, SyntaxToken eof) {
```

Comment:
This is not a Java Optional. It's an SSLR Optional. The NodeBuilder will return this kind of Optional here. Unfortunately, not easy to avoid this without changing SSLR itself.

Reply:
Oh okay, sorry, I didn't notice it's SSLR Optional. Then please discard my comment.
However I wonder if we should change `b.zeroOrMore(DOCKERIMAGE())` to `b.oneOrMore(DOCKERIMAGE())`. My understanding is that it's mandatory to have at least one "FROM ... "